README.md

```diff
@@ -114,6 +114,11 @@ This library automatically garbage collects all C objects created through its AP
 
 ## Concurrency
 
-If you intend to use this library's API from multiple threads,
-you must use a unique `Z3.Base.Context` object per thread.
-
+Since version `408.3`, this library implements thread-safety over the C API,
+i.e. API calls are serialized by locking on their `Context` argument.
+To safely compile for multi-threaded code please upgrade to `>= 408.3`.
+
+Operations and objects in different `Context`s can safely be accessed concurrently
+and are not synchronized by this library.
+So if you are creating multiple threads using this library you should use a unique `Context` per thread.
```

Comment:
Sorry for the late feedback. This one line I don't understand well, once we serialize all uses of a given `Context`, why do the users need to use a unique `Context` per thread?

Reply:
It's definitely not worded well. I am trying to say that if you want to achieve real concurrency between your own threads you must use distinct `Context`'s, otherwise you will get serialized. I can also simply remove the last sentence and it should be inferrable from the rest.

Reply:
Now I understand your point. You can explain it the way you just did. Something like:

> Operations on objects from different `Context`s can safely be accessed concurrently without synchronization. Therefore, if you want to achieve real concurrency, you must use a different `Context` in each thread. You can use the _translate_ functions to copy objects between different `Context`s.